core/src/env/api.rs

```diff
@@ -19,20 +19,23 @@ use crate::{
     env::{
         Env as _,
         EnvStorage as _,
-        EnvTypes,
     },
     memory::vec::Vec,
     storage::Key,
 };
+use node_runtime::{
+    self,
+    Runtime,
+};
 
 /// The environmental address type.
-pub type AccountId = <ContractEnv as EnvTypes>::AccountId;
+pub type AccountId = <Runtime as srml_system::Trait>::AccountId;
 
 /// The environmental balance type.
-pub type Balance = <ContractEnv as EnvTypes>::Balance;
+pub type Balance = <Runtime as srml_balances::Trait>::Balance;
```

Comment:
could this be `srml_contract::BalanceOf<Runtime>` so it is more compatible with our runtime that doesn't use srml_balances module.

Reply:
I agree here. The only module ink should depend on is the `contract` module. Any other types or dependencies need to come through the `contract` module to ensure that people can really customize their substrate chain in a modular way.

Reply:
Types in the `contract` module require an implementation of the `contract::Trait` which is why we import `node-runtime`.

I agree that this is suboptimal since chains will not necessarily have the same runtime module construction.

So we need to figure out the best way to provide a default `Runtime` but also allow a user supplied custom `Runtime`